lib/lunar.js

```diff
@@ -163,36 +168,46 @@ function altitude_julian(h, event, t, lat, lon) {
     }
 
     if (roots === 1) {
-      if (h0 < 0) { 
-        times.rise = i + x1;
-      } else {
-        times.set = i + x1;
+      if (h0 < 0 && isNaN(times.moonrise)) {
+        times.moonrise = i + x1;
+      } else if(isNaN(times.moonset)) {
+        times.moonset = i + x1;
       }
     } else if (roots === 2) {
-      times.rise = i + (ye < 0 ? x2 : x1);
-      times.set  = i + (ye < 0 ? x1 : x2);
+      times.moonrise = isNaN(times.moonrise) ?
+          i + (ye < 0 ? x2 : x1) : times.moonrise;
+      times.moonset  = isNaN(times.moonset) ?
+          i + (ye < 0 ? x1 : x2): times.moonset;
```

Comment:
These `?:` conditionals feel a little weird to me. Why not just `if(!isNaN(times.moonrise)) { times.moonrise = ...; }`? (That is, if it's NaN, we can just leave it alone!)